pkg/reconciliation/reconcile_racks.go

```diff
@@ -672,6 +672,10 @@ func (rc *ReconciliationContext) checkSeedLabels() (int, error) {
 	return seedCount, nil
 }
 
+func shouldUseFastPath(dc *api.CassandraDatacenter, seedCount int) bool {
+	return seedCount > 0 && !(metav1.HasAnnotation(dc.ObjectMeta, api.AllowParallelStartsAnnotations) && dc.Annotations[api.AllowParallelStartsAnnotations] == "false")
```

Comment:
I'm suggesting:

```
return seedCount > 0 && 
  !metav1.HasAnnotation(dc.ObjectMeta, api.AllowParallelStartsAnnotations) &&
  !dc.Annotations[api.AllowParallelStartsAnnotations] == "false"
```

Reply:
That code wouldn't even compile. What you probably meant was:

```
return seedCount > 0 && 
  !metav1.HasAnnotation(dc.ObjectMeta, api.AllowParallelStartsAnnotations) &&
  dc.Annotations[api.AllowParallelStartsAnnotations] != "false"
```

And that changes the logic. Not just because dc.Annotations could be nil in this case (as HasAnnotation checks it's not nil or it would return false, yet now we get false false -> true), but also because the check is different. The test even revealed this.

If dc.Annotations[api.AllowParallelStartsAnnotations] == "true", then we should use the fastpath. The code you proposed returns false in this case, since the annotation has been set and in that case the proposed code would return false. And it's allowed to be set, we should only reject the fastpath if it's set to false, no other case.